rest_framework/urlpatterns.py

```diff
@@ -99,14 +99,14 @@ def format_suffix_patterns(urlpatterns, suffix_required=False, allowed=None):
         if len(allowed) == 1:
             allowed_pattern = allowed[0]
         else:
-            allowed_pattern = '(%s)' % '|'.join(allowed)
+            allowed_pattern = f'({"|".join(allowed)})'
```

Comment:
We could do the same here.

```suggestion
            allowed_cases = "|".join(allowed)
            allowed_pattern = f'({allowed_cases})'
```